internal/service/app/query/get_game_details_mock.go

```diff
@@ -0,0 +1,36 @@
+package query
+
+import (
+	"github.com/nhood-org/gre/internal/service/app/model"
+)
+
+type GameDetailsRepositoryMock struct {
+	CapturedUUIDs* []model.GameDetailsUUID
+	CaptureGetByInParam* string
+	OutGame model.GameDetails
+	OutGames []model.GameDetails
+}
+
+type SearchEngineMock struct {
+	MetadataCaptured* model.MetadataVector
+	GamesUUIDs []model.GameDetailsUUID
+}
+
+func (g GameDetailsRepositoryMock) GetByUUID(uuid model.GameDetailsUUID) (model.GameDetails, error) {
+	*g.CapturedUUIDs = append(*g.CapturedUUIDs, uuid)
+	return g.OutGame, nil
+}
```

Comment:
The better practice is to use pointer receivers so that everytime the method is called you have access to the very same instance:
```suggestion
func (g *GameDetailsRepositoryMock) GetByUUID(uuid model.GameDetailsUUID) (model.GameDetails, error) {
	g.CapturedUUIDs = append(g.CapturedUUIDs, uuid)
	return g.OutGame, nil
}
```

Reply:
thanks for the hint! done